docs/dev/development-process/development-process.rst

```diff
@@ -0,0 +1,209 @@
+    :caption: Artemis/.github/ISSUE_TEMPLATE/feature-proposal--developer-.md
+    :language: markdown
+
+
+Step 2: Requirements Engineering
+--------------------------------
+In this section, the foundation for the feature's development is laid by meticulously defining both functional and non-functional requirements.
+Functional requirements detail what the feature will do, covering specific behaviors or functions.
+Non-functional requirements, on the other hand, address the feature's operation, such as performance, usability, reliability, and security specifications.
+
+In the Feature Proposal Template, you can find examples for structuring such functional and non-functional requirements
+
+Step 3: Analysis
+----------------
+Within the analysis section, the emphasis is on understanding and documenting the feature's intended behavior and interactions within the system.
+This includes creating an Analysis Object Model, which represents the static structure of the system using class diagrams to identify key entities and their relationships.
+Additionally, dynamic aspects of the feature are explored through Activity Diagrams (for workflow representation), State Chart Diagrams (to model the feature's states and transitions), and Communication Diagrams (illustrating how objects interact to achieve the feature's functionality).
+These models serve as a blueprint for the development and ensure a shared understanding among all people working on Artemis.
+
+Step 4: System Architecture
+---------------------------
+In this step, the high-level structure of the system supporting the feature is outlined, focusing on analyzing the system components relevant to the implementation of the feature,
+subsystem decomposition, interfaces, persistent data management, and security aspects.
+The subsystem decomposition process is used to identify the main components of the system and define their responsibilities, which facilitates modular development and maintenance.
+It is important to define the interfaces between these subsystems to ensure seamless interaction and integration.
+In addition, the development of a comprehensive data model is important for a clear understanding of how data is securely stored, retrieved, and managed within the system, taking into account both logical and physical data perspectives.
+Security aspects, especially access control mechanisms, are considered to ensure data integrity and privacy.
+
+Step 5: UI/UX Design
+--------------------
+If the proposed feature requires a change to the software's user interface, this section should include a detailed description of the proposed changes, as well as a mockup that illustrates the new user interface.
+The mockup should be created using Figma in conjunction with the [Artemis Design System](https://www.figma.com/files/team/1238833580691926997/Artemis) and should be added to the feature proposal as a screenshot.
```

Comment:
I suspect this link is in the wrong format?
<img width="990" alt="Screenshot 2024-05-08 at 11 30 45" src="https://github.com/ls1intum/Artemis/assets/13572283/a331989e-4479-4e4d-a658-1f1275b8aa12">